shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPreparer.java

```diff
@@ -142,10 +141,19 @@ private void prepareTarget(final RuleAlteredJobContext jobContext) {
             log.info("dataSourcePreparer null, ignore prepare target");
             return;
         }
+        DatabaseType sourceDatabaseType = DatabaseTypeFactory.getInstance(jobConfig.getSourceDatabaseType());
+        DatabaseType targetDatabaseType = DatabaseTypeFactory.getInstance(jobConfig.getTargetDatabaseType());
+        if (!sourceDatabaseType.isSchemaAvailable() || !targetDatabaseType.isSchemaAvailable()) {
+            log.info("prepareTargetSchemas, one of source or target database type schema is not available, ignore");
+            return;
+        }
```

Comment:
Looks it could not return, since it just could ignore creating schema but not creating table.

Could we add an judgement method to check whether it need to create schema or not? And then use it to precheck for `prepareTargetSchemas`.

Reply:
Could you tell me the idea for the method？

Reply:
For example:
```
private boolean isSourceAndTargetSchemaAvailable(... jobConfig) {
...
}
```
Then it could be used in `prepareTarget`:
```
if (isSourceAndTargetSchemaAvailable(...)) {
    dataSourcePreparer.get().prepareTargetSchemas(prepareTargetSchemasParameter);
}
```